src/Node/Stream.purs

```diff
@@ -61,6 +65,18 @@ type Writable r = Stream (write :: Write | r)
 -- | A duplex (readable _and_ writable stream)
 type Duplex = Stream (read :: Read, write :: Write)
 
+data Chunk
+data Data = BufferData Buffer
+          | StringData String
+          | UnknownData Chunk
```

Comment:
My impression is that object mode lets you put any javascript value through a stream (apart from `null` which has a special meaning for the end of the stream). Given this, I think `Foreign` is appropriate; that is what the type represents, and you also get plenty of useful functions for turning `Foreign` values into data types that you can do stuff with.

Why do you say `Foreign` is more ambiguous than your `Chunk`?

Looking at the node API again though, if we were going to support object mode, I think I would rather have an entirely separate data type for streams in object mode anyway, especially since no Node APIs use them. I don't want to have to worry about whether a stream is in object mode when it ought to be possible to rule that out at compile time.